cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/saved_search.spec.js

```diff
@@ -0,0 +1,513 @@
+    saveName: 'dql-index-pattern-01',
+  },
+  Lucene: {
+    ...allowedSearchOperations.Lucene,
+    language: 'Lucene',
+    dataset: INDEX_PATTERN_WITH_TIME,
+    queryString: 'bytes_transferred: {9950 TO *}',
+    hitCount: 28,
+    sampleTableData: [
+      [1, '9,998'],
+      [2, 'Phyllis Dach'],
+    ],
+    saveName: 'lucene-index-pattern-01',
+  },
+  'OpenSearch SQL': {
+    ...allowedSearchOperations['OpenSearch SQL'],
+    language: 'OpenSearch SQL',
+    dataset: INDEX_PATTERN_WITH_TIME,
+    queryString: `SELECT * FROM ${INDEX_PATTERN_WITH_TIME} WHERE bytes_transferred > 9950`,
+    hitCount: undefined,
+    sampleTableData: [],
+    saveName: 'sql-index-pattern-01',
+  },
+  PPL: {
+    ...allowedSearchOperations.PPL,
+    language: 'PPL',
+    dataset: INDEX_PATTERN_WITH_TIME,
+    queryString: `source = ${INDEX_PATTERN_WITH_TIME} | where bytes_transferred > 9950`,
+    hitCount: 101,
+    sampleTableData: [],
+    saveName: 'ppl-index-pattern-01',
+  },
+};
+const indexTestConfigurations = {
+  'OpenSearch SQL': {
+    ...allowedSearchOperations['OpenSearch SQL'],
+    language: 'OpenSearch SQL',
+    dataset: INDEX_WITH_TIME_1,
+    queryString: `SELECT * FROM ${INDEX_WITH_TIME_1} WHERE bytes_transferred > 9950`,
+    hitCount: undefined,
+    sampleTableData: [],
+    saveName: 'sql-index-01',
+  },
+  PPL: {
+    ...allowedSearchOperations.PPL,
+    language: 'PPL',
+    dataset: INDEX_WITH_TIME_1,
+    queryString: `source = ${INDEX_WITH_TIME_1} | where bytes_transferred > 9950`,
+    hitCount: 50,
+    sampleTableData: [],
+    saveName: 'ppl-index-01',
+  },
+};
+const allTestConfigurations = [
+  ...Object.values(indexPatternTestConfigurations),
+  ...Object.values(indexTestConfigurations),
+];
+
+// Maps a dataset that are used in this file to the exact string that dataset
+// corresponds to in the saved search API response
+const mapDatasetToType = (dataset) => {
+  switch (dataset) {
+    case INDEX_PATTERN_WITH_TIME:
+      return 'INDEX_PATTERN';
+    case INDEX_WITH_TIME_1:
+      return 'INDEXES';
+    default:
+      return 'unknown dataset';
+  }
+};
+
+// Maps a language that are used in this file to the exact string that the language
+// corresponds to in the saved search API response
+const mapLanguageToApiResponseString = (language) => {
+  switch (language) {
+    case 'DQL':
+      return 'kuery';
+    case 'Lucene':
+      return 'lucene';
+    case 'OpenSearch SQL':
+      return 'SQL';
+    case 'PPL':
+      return 'PPL';
+    default:
+      return 'unknown language';
+  }
+};
+
+// Escapes special characters for the editor
+const prepareQueryStringForEditor = (queryString) => {
+  return queryString.replaceAll(/([{}])/g, (char) => `{${char}}`);
+};
+
+// In theory this function is not needed as we should be able to use
+// the custom command navigateToWorkSpaceSpecificPage, but when using that
+// to go to the discover page, it sometimes leads to a blank page
+const navigateToDiscoverPage = () => {
```

Comment:
nit: Can we move this logic to the global function it can help to make that command more robust

Reply:
I refactored this so that this function is no longer needed